File: backend/src/main/java/com/decmoe47/todo/model/entity/Todo.java
package com.decmoe47.todo.model.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

import java.time.LocalDateTime;

@EqualsAndHashCode(callSuper = true)
@Accessors(chain = true)
@Data
@Entity
public class Todo extends Auditable<User> {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String content;
    private LocalDateTime dueDate;
    private boolean done = false;
    private String description;

    @ManyToOne
    @JoinColumn(name = "belonged_list_id")
    private TodoList belongedList;
}

File: backend/src/main/kotlin/com/decmoe47/todo/model/entity/Todo.kt
package com.decmoe47.todo.model.entity

import org.komapper.annotation.KomapperAutoIncrement
import org.komapper.annotation.KomapperEmbedded
import org.komapper.annotation.KomapperEntity
import org.komapper.annotation.KomapperId
import java.time.LocalDateTime

@KomapperEntity
data class Todo(
    @KomapperId @KomapperAutoIncrement
    val id: Long = 0,
    val content: String,
    val dueDate: LocalDateTime?,
    val done: Boolean = false,
    val description: String? = null,
    val belongedListId: Long,
    @KomapperEmbedded
    val auditable: AuditableEntity
) : Auditable by auditable
